panel/models/tabulator.ts

```diff
@@ -1070,6 +1070,28 @@ export class DataTabulatorView extends HTMLBoxView {
     let indices: number[] = []
     const selected = this.model.source.selected
     const index: number = row._row.data._index
+
+    if (this.model.pagination === 'remote') {
+      let includes = this.model.source.selected.indices.indexOf(index) == -1
+      let flush = !(e.ctrlKey || e.metaKey || e.shiftKey)
+      if (e.shiftKey && selected.indices.length) {
+        const start = selected.indices[selected.indices.length-1]
+        if (index>start) {
+          for (let i = start; i<=index; i++)
```

Comment:
In the non-remote logic, this is done with `for (let i = start; i<index; i++)`. I feel like it makes more sense to go up to the selected index and not one above/below it.

Reply:
Indeed, was that a bug?

Reply:
I think it is. It was added in #2230.

Reply:
I have updated it also to be the case for the non-remote option.

Reply:
It was not a bug, have reverted the change. The index is added in the following if statement:

https://github.com/holoviz/panel/blob/078591fda4150421c1601217ae54a9b4d321cf4f/panel/models/tabulator.ts#L1107-L1109